packages/@react-spectrum/s2/src/Table.tsx

```diff
@@ -11,85 +11,1082 @@
+      layoutInfo.rect = new Rect(40, 40, this.virtualizer.visibleRect.width - 80, this.virtualizer.visibleRect.height - 80);
+      layoutInfo.isSticky = true;
+    }
+
+    return {...layoutNode, layoutInfo};
+  }
+
+  protected buildRow(node: GridNode<T>, x: number, y: number): LayoutNode {
+    let layoutNode = super.buildRow(node, x, y);
+    layoutNode.layoutInfo.allowOverflow = true;
+    // Needs overflow for sticky selection/drag cells
+    return layoutNode;
+  }
+
+  protected buildTableHeader(): LayoutNode {
+    let layoutNode = super.buildTableHeader();
+    // Needs overflow for sticky selection/drag column
+    layoutNode.layoutInfo.allowOverflow = true;
+    return layoutNode;
+  }
+
+  protected buildColumn(node: GridNode<T>, x: number, y: number): LayoutNode {
+    let layoutNode = super.buildColumn(node, x, y);
+    // Needs overflow for the resize handle
+    layoutNode.layoutInfo.allowOverflow = true;
+    return layoutNode;
+  }
+}
 
 export function Table(props: TableProps) {
-  return <AriaTable {...props} />;
+  let {
+    UNSAFE_style,
+    UNSAFE_className,
+    isQuiet = false,
+    density = 'regular',
+    overflowMode = 'truncate',
+    styles,
+    loadingState,
+    onLoadMore,
+    onResize: propsOnResize,
+    onResizeStart: propsOnResizeStart,
+    onResizeEnd: propsOnResizeEnd,
+    onAction,
+    ...otherProps
+  } = props;
+
+  let scale = (useIsMobileDevice() ? 'large' : 'medium') as Scale;
+  let layout = useMemo(() => {
+    return new S2TableLayout({
+      rowHeight: overflowMode === 'wrap'
+        ? undefined
+        : ROW_HEIGHTS[density][scale],
+      estimatedRowHeight: overflowMode === 'wrap'
+      ? ROW_HEIGHTS[density][scale]
+      : undefined,
+      // No need for estimated headingHeight since the headers aren't affected by overflow mode: wrap
+      headingHeight: DEFAULT_HEADER_HEIGHT[scale]
+    });
+  }, [overflowMode, density, scale]);
+
+  // Starts when the user selects resize from the menu, ends when resizing ends
+  // used to control the visibility of the resizer Nubbin
+  let [isInResizeMode, setIsInResizeMode] = useState(false);
+  let onResizeStart = useCallback((widths) => {
+    propsOnResizeStart?.(widths);
+  }, [propsOnResizeStart]);
+  let onResizeEnd = useCallback((widths) => {
+    setIsInResizeMode(false);
+    propsOnResizeEnd?.(widths);
+  }, [propsOnResizeEnd, setIsInResizeMode]);
+
+  let context = useMemo(() => ({
+    isQuiet,
+    density,
+    overflowMode,
+    loadingState,
+    isInResizeMode,
+    setIsInResizeMode,
+    scale,
+    layout
+  }), [isQuiet, density, overflowMode, loadingState, scale, layout, isInResizeMode, setIsInResizeMode]);
+
+  let isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
+  let scrollRef = useRef(null);
+  let memoedLoadMoreProps = useMemo(() => ({
+    isLoading: isLoading,
+    onLoadMore
+  }), [isLoading, onLoadMore]);
+  useLoadMore(memoedLoadMoreProps, scrollRef);
```

Comment:
This may change when we revisit `useLoadMore` to possibly accept collections, but one thing to note is that the `items` aren't available here since they are provided directly to the TableBody instead of the Table. This causes the issue described in https://github.com/orgs/adobe/projects/19/views/32?sliceBy%5Bvalue%5D=S2&pane=issue&itemId=79185605 where a new load more isn't triggered if items are removed from view manually. We could perhaps add the `items` prop to the Table as well but I'm not sure what form `useLoadMore` to support `collections` so I'm punting for now

Reply:
Same problem in cardview. We will have to move it into RAC to handle this.